mlir/lib/IR/MLIRContext.cpp

```diff
@@ -212,6 +212,9 @@ class MLIRContextImpl {
   DenseMap<TypeID, AbstractType *> registeredTypes;
   StorageUniquer typeUniquer;
 
+  /// This is a mapping from type name to the abstract type describing it.
+  llvm::StringMap<AbstractType *> nameToType;
```

Comment:
Can we use `DenseMap<StringAttr, AbstractType *> nameToType;` (more efficient storage, and more efficient lookup)?

Also when is it useful? (please add to the comment: the "what" is less important than the "why").

Reply:
Added!

Reply:
did you?

Reply:
Oh my bad, it got erased again in a following PR, I'll add it back now!
Sorry about that!